operations/pyroscope/helm/pyroscope/templates/NOTES.txt

```diff
@@ -1,19 +1,50 @@
-Thanks for deploying Grafana Phlare.
+Thanks for deploying Grafana Pyroscope.
 
-In order to configure Grafana to use the Phlare datasource, you need to add the Phlare datasource to your Grafana instance.
+# Pyroscope UI & Grafana
 
-The in-cluster query URL is:
+Pyroscope database comes with a built-in UI, to access it from your localhost you can use:
 
+```
 {{- if hasKey .Values.pyroscope.components "query-frontend" }}
-http://{{ include "pyroscope.fullname" . }}-query-frontend.{{ .Release.Namespace }}.svc.cluster.local.:{{ .Values.pyroscope.service.port }}
+kubectl --namespace {{ .Release.Namespace }} port-forward svc/{{ include "pyroscope.fullname" . }}-query-frontend {{ .Values.pyroscope.service.port }}:{{ .Values.pyroscope.service.port }}
 {{- else }}
-http://{{ include "pyroscope.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local.:{{ .Values.pyroscope.service.port }}
+kubectl --namespace {{ .Release.Namespace }} port-forward svc/{{ include "pyroscope.fullname" . }} {{ .Values.pyroscope.service.port }}:{{ .Values.pyroscope.service.port }}
 {{- end }}
+```
+
+You can also use Grafana to explore Pyroscope data.
+For that, you'll need to add the Pyroscope data source to your Grafana instance and configure the query URL accordingly.
+See https://grafana.com/docs/grafana/latest/datasources/grafana-pyroscope/ for more details.
 
-To forward the query API to your localhost you can use:
+The in-cluster query URL for the data source in Grafana is:
 
+```
 {{- if hasKey .Values.pyroscope.components "query-frontend" }}
-kubectl --namespace {{ .Release.Namespace }} port-forward svc/{{ include "pyroscope.fullname" . }}-query-frontend {{ .Values.pyroscope.service.port }}:{{ .Values.pyroscope.service.port }}
+http://{{ include "pyroscope.fullname" . }}-query-frontend.{{ .Release.Namespace }}.svc.cluster.local.:{{ .Values.pyroscope.service.port }}
```

Comment:
This port is wrong when you override the port only for that one component.

You would need to do the default thing here, too.

Reply:
Yeah not easy with the dash. :thinking:

Reply:
```
{{ $port := ((index .Values.pyroscope.components "query-frontend").service).port | default .Values.pyroscope.service.port }}
```

did the trick :mindblowing: